### DISCUSSION TITLE
FDS HRR DEVC setup error: "FDS cannot determine which boundary cell to assign."

### DISCUSSION THREAD

**Discussion by darwiethepang02** [firemodels/fds#12799] [Q&A]
Hello, I would like to ask how do I solve the error when setting up a devc to collect data for heat release rate as I keep facing the error "FDS cannot determine which boundary cell to assign."? I have tried changing the coordinates slightly between the boundary range as initially I placed on the maximum and minimum coordinates of the space but still unable to solve it. Another method I have tried is to play around with the mesh size but still unable to solve the error.

**Comment by Er9y714** [comment]
Have you tried adding orientation to the device? (IOR)

**Reply by darwiethepang02** [reply]
Not yet but will have a try now. Thank you

**Reply by darwiethepang02** [reply]
I have tried adding in IOR = 1.0 as a test to the code but still having the same error.

**Reply by Er9y714** [reply]
Can you share the device line becuase I am not sure how you are measuring HRR with a boundary requiring device.

**Reply by darwiethepang02** [reply]
The conditions are: 
Room dimensions: 6.26 x 5.26 x 2.4 m including brick wall thickness of 260mm 
Fire source is placed in the middle of the office space with a dimension of 0.9 x 0.9 x 0.792 m 

Coding for the device is as below: 
&DEVC XYZ = 3.13, 2.63, 0.01, QUANTITY = 'NORMALIZED HEAT RELEASE RATE', ID = 'HRR_1', IOR = -1.0/
&DEVC XYZ = 3.13, 2.63, 1.3, QUANTITY = 'NORMALIZED HEAT RELEASE RATE', ID = 'HRR_2', IOR = -1.0/
&DEVC XYZ = 3.13, 2.63, 2.3, QUANTITY = 'NORMALIZED HEAT RELEASE RATE', ID = 'HRR_3', IOR = -1.0/

The IOR I have tried 1.0 and -1.0 but both have the same error. 
Also, I am unsure if the quantity is correct or not so do correct me if there is a mistake.

Thank you.

**Comment by Er9y714** [comment]
I do not have experience with the quantity you are measuring but IOR should be set in according to the surface you are measuring. If the normal is +x, IOR should be 1, for -x : -1, +y : 2, -y : -2, +z : 3, -z: -3. 

Based on the IOR you have set I understand the surface you are measuring is a vertical wall and it faces -x direction.  Is that correct?

**Reply by darwiethepang02** [reply]
I am not familiar with how the IOR works so I just applied both 1.0 and -1.0 value. Is there a way to measure the HRR without needing the IOR function because my goal of this project is to measure the HRR of the fire source.

**Reply by Er9y714** [reply]
You can find the calculated HRR in _hrr.csv file, the first column.

**Reply by darwiethepang02** [reply]
But that one is for the room itself or the fire source?

**Reply by Er9y714** [reply]
_hrr.csv file will you the total of the domain. If you would like to limit to a certain volume, you can try the example below

&DEVC ID='HRR',  XB=0.0,1.0,0.0,1.0,0,1.0, QUANTITY='HRRPUV', SPATIAL_STATISTIC='VOLUME INTEGRAL',/

**Reply by darwiethepang02** [reply]
Alright thank you. Will have a try now.

**Reply by darwiethepang02** [reply]
With this code right, do I view the data in the same _hrr.csv.file?

**Reply by Er9y714** [reply]
This one is a &DEVC line therefore it is a device. You will find the outputs in _devc.csv file. 
I recommend reading the user guide how devices work, output interval is set and output structure.

**Reply by darwiethepang02** [reply]
Alright thank you. Yes I think this is the code I needed. Thank you very much for the help and advice.
